File: trunk/mobileCloud/android/2_0/test-suite/test-sync-port/src/main/java/org/openmobster/core/mobileCloud/android/module/sync/TestObjectStreaming.java
/******************************************************************************
 * OpenMobster                                                                *
 * Copyright 2008, OpenMobster, and individual                                *
 * contributors as indicated by the @authors tag. See the                     *
 * copyright.txt in the distribution for a full listing of                    *
 * individual contributors.                                                   *
 *                                                                            *
 * This is free software; you can redistribute it and/or modify it            *
 * under the terms of the GNU Lesser General Public License as                *
 * published by the Free Software Foundation; either version 2.1 of           *
 * the License, or (at your option) any later version.                        *
 *                                                                            *
 * This software is distributed in the hope that it will be useful,           *
 * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU           *
 * Lesser General Public License for more details.                            *
 *                                                                            *
 * You should have received a copy of the GNU Lesser General Public           *
 * License along with this software; if not, write to the Free                *
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA         *
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.                   *
 ******************************************************************************/
package org.openmobster.core.mobileCloud.android.module.sync;

import org.openmobster.core.mobileCloud.android.module.mobileObject.MobileObject;


/**
 * 
 * @author openmobster@gmail.com
 *
 */
public final class TestObjectStreaming extends AbstractSyncTest 
{		
	/**
	 * 
	 */
	public void runTest()
	{
		try
		{
			this.setUp("add");
			SyncService.getInstance().performTwoWaySync(service, service, false);					
			this.assertRecordPresence("unique-1", "/TestTwoWaySync/add");
			this.assertRecordPresence("unique-2", "/TestTwoWaySync/add");
			this.assertRecordPresence("unique-3", "/TestTwoWaySync/add");
			this.assertRecordPresence("unique-4", "/TestTwoWaySync/add");
			
			MobileObject unique1 = this.getRecord("unique-1");
			this.assertNull(unique1.getValue("attachment"), "/TestObjectStreaming/NoAttachmentDownloaded");
			
			SyncService.getInstance().performStreamSync(service, "unique-1", false);
			unique1 = this.getRecord("unique-1");
			this.assertNotNull(unique1.getValue("attachment"), "/TestObjectStreaming/AttachmentMustBeDownloaded");
			
			this.tearDown();
						
		}
		catch(Exception e)
		{
			e.printStackTrace(System.out);
			throw new RuntimeException(e.toString());
		}
	}	
}

File: iphone/mobilecloudlib/TestObjectStreaming.m
/**
 * Copyright (c) {2003,2010} {openmobster@gmail.com} {individual contributors as indicated by the @authors tag}.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */

#import "TestObjectStreaming.h"


/**
 * 
 * @author openmobster@gmail.com
 */
@implementation TestObjectStreaming

+(id)withInit
{
	return [[[TestObjectStreaming alloc] init] autorelease];
}

-(void) runTest
{
	SyncService *sync = [SyncService getInstance];
	
	[self setUp:@"add"];
	[sync performTwoWaySync:channel :NO];
	[self assertRecordPresence:@"unique-1" :@"/TestObjectStreaming/add"];
	[self assertRecordPresence:@"unique-2" :@"/TestObjectStreaming/add"];
	[self assertRecordPresence:@"unique-3" :@"/TestObjectStreaming/add"];
	[self assertRecordPresence:@"unique-4" :@"/TestObjectStreaming/add"];
	
	MobileObject *unique1 = [self getRecord:@"unique-1"];
	NSString *attachment = [unique1 getValue:@"attachment"];
	[self assertTrue:attachment == nil :@"/TestObjectStreaming/AttachmentMustNotBeDownloaded"];
	
	
	[sync performStreamSync:channel :NO :@"unique-1"];
	unique1 = [self getRecord:@"unique-1"];
	attachment = [unique1 getValue:@"attachment"];
	[self assertTrue:attachment != nil :@"/TestObjectStreaming/AttachmentMustBeDownloaded"];
	
	[self tearDown];
}

-(NSString *) getInfo
{
	return [[self class] description];
}
@end
